cpp/open3d/t/geometry/TriangleMesh.cpp

```diff
@@ -174,6 +174,98 @@ TriangleMesh &TriangleMesh::Rotate(const core::Tensor &R,
     return *this;
 }
 
+TriangleMesh &TriangleMesh::RemoveDuplicatedTriangles() {
+    core::Tensor triangles = GetTriangleIndices();
+    int64_t num_triangles = triangles.GetLength();
+    if (num_triangles < 1) {
+        utility::LogError("Mesh must have at least two triangles.");
+    }
+
+    // unordered_map to keep track of existing triangles
+    // hash function with code::Tensor not implemented so we use std::tuple
+    typedef std::tuple<int, int, int> Index3;
+    std::unordered_map<Index3, size_t, utility::hash_tuple<Index3>>
+            triangle_to_old_index;
+
+    // Check is mesh has normals triangles
+    bool has_tri_normal = HasTriangleNormals();
+    // index of triangle to keep (unique triangle)
+    std::vector<int64_t> index_to_keep;
+    int64_t new_size = 0;
+    for (int64_t i = 0; i < num_triangles; i++) {
+        Index3 triangle_in_tuple;
+        // Extract vertices of the triangle
+        core::Tensor triangle_to_check =
+                triangles.GetItem({core::TensorKey::Index(i)});
+        int64_t vertice_0 =
+                triangle_to_check.GetItem({core::TensorKey::Index(0)})
+                        .Item<int64_t>();
+        int64_t vertice_1 =
+                triangle_to_check.GetItem({core::TensorKey::Index(1)})
+                        .Item<int64_t>();
+        int64_t vertice_2 =
+                triangle_to_check.GetItem({core::TensorKey::Index(2)})
+                        .Item<int64_t>();
```

Comment:
We need to ensure this works for different dtypes for vertices (specifically int32_t and int64_t). This will output junk values for int32_t vertices.

**minor:**

- `operator[]` is shorter: `triangle_to_check[0].Item<dtype>()`
- Would using `std::rotate` to reorder elements in the tuple be cleaner than the multiple `ifs` ?
- Can you comment on parallelization here? Is it possible and would it provide a useful speedup?

Reply:
std::rotate can be used but it will need multiple ifs too to know which vertices to rotate. Example {2,1,0} will need 2 rotations to get {0,1,2}. What can be used is instantiate a std::vector<int64_t> of 3 elements, sort it with std::sort and next initialize the tuple for the unordered map.
Also elements of triangle_vec are int64_t like that it can get vertices in int32_t or int64_t.
```
        std::vector<int64_t> triangle_vec;
        if (triangles.GetDtype() == core::Int32)
        {
            auto vertice_0 = triangle_to_check[0].Item<int32_t>();
            auto vertice_1 = triangle_to_check[1].Item<int32_t>();
            auto vertice_2 = triangle_to_check[2].Item<int32_t>();
            triangle_vec = {vertice_0, vertice_1, vertice_2};
        }
        else if (triangles.GetDtype() == core::Int64)
        {
            auto vertice_0 = triangle_to_check[0].Item<int64_t>();
            auto vertice_1 = triangle_to_check[1].Item<int64_t>();
            auto vertice_2 = triangle_to_check[2].Item<int64_t>();
            triangle_vec = {vertice_0, vertice_1, vertice_2};
        }

        // We first need to find the minimum index. Because triangle (0-1-2)
        // and triangle (2-0-1) are the same.
        std::sort(triangle_vec.begin(), triangle_vec.end());
        triangle_in_tuple =
                        std::make_tuple(triangle_vec[0], triangle_vec[1], triangle_vec[2]);
```

The parallelization is possible but we need to be sure that the access to the unordered map is in a critical section to avoid multiple accesses adding the duplicate triangle in the unordered map.

Reply:
done in https://github.com/isl-org/Open3D/pull/6409/commits/c890dc81096ad395074fad5d02cc813c7cd789bf